src/app/services/actions/permissions-action-creator.ts

```diff
@@ -155,3 +197,140 @@ export function consentToScopes(scopes: string[]) {
     }
   };
 }
+
+interface IPreliminaryChecksObject {
+  defaultUserScopes: string[];
+  requiredPermissions: string[];
+  consentedScopes: string[];
+  permissionToRevoke: string;
+  grantsPayload?: IOAuthGrantPayload;
+}
+
+enum REVOKE_STATUS {
+  success = 'success',
+  failure = 'failure',
+  preliminaryChecksFail = 'preliminaryChecksFail',
+  allPrincipalScope = 'allPrincipalScope'
+}
+
+export function revokeScopes(permissionToRevoke: string) {
```

Comment:
This function looks too long. Is there a way for us to break it down more so that it's easier to read?